imjoy/worker_template.py

```diff
@@ -70,86 +62,53 @@ def ndarray(typed_array, shape, dtype):
 class PluginConnection:
     """Represent a plugin connection."""
 
-    def __init__(
-        self,
-        pid,
-        secret,
-        server,
-        job_queue=None,
-        loop=None,
-        worker=None,
-        work_dir=None,
-        daemon=False,
-    ):
-        """Set up connection."""
-        if work_dir is None or work_dir == "" or work_dir == ".":
-            self.work_dir = os.getcwd()
-        else:
-            self.work_dir = work_dir
-            if not os.path.exists(self.work_dir):
-                os.makedirs(self.work_dir)
-            os.chdir(self.work_dir)
-        socket_io = SocketIO(server, Namespace=LoggingNamespace)
-        self.socket_io = socket_io
-        self.init = False
-        self.secret = secret
-        self.id = pid  # pylint: disable=invalid-name
-        self.daemon = daemon
-
-        def emit(msg):
-            socket_io.emit("from_plugin_" + secret, msg)
-
-        self.emit = emit
+    # pylint:disable=too-many-instance-attributes
 
+    def __init__(self, opt):
+        """Set up connection instance."""
+        self.secret = opt.secret
+        self.id = opt.id  # pylint: disable=invalid-name
         self.local = {}
         self._remote = dotdict()
-        self._set_local_api(self._remote)
         self.interface = {}
         self.plugin_interfaces = {}
         self.remote_set = False
         self.store = ReferenceStore()
         self.executed = False
-        self.queue = job_queue
-        self.loop = loop
-
         self.init = False
-        sys.stdout.flush()
-        socket_io.on("to_plugin_" + secret, self.sio_plugin_message)
-        self.emit({"type": "initialized", "dedicatedThread": True})
-        logger.info("Plugin %s initialized", pid)
-
-        def on_disconnect():
-            if not self.daemon:
-                self.exit(1)
-
-        socket_io.on("disconnect", on_disconnect)
         self.abort = threading.Event()
-        self.worker = worker
-        self.sync_q = None
-
-    def wait_forever(self):
-        """Wait forever."""
+        self.work_dir = opt.work_dir
+        self.opt = opt
         if PYTHON3:
```

Comment:
Should we completely separate python3 and python 2 code here?

Reply:
We have to check somewhere for python version to know what client to use. Where should we do that?

Reply:
Not sure exactly, but I was thinking whether we should have separate worker_template for both, it seems that would not make much sense. Never mind, let's keep it like this.